libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -267,6 +271,52 @@ instance HasDeprecatedFeatureName 'TeamFeatureDigitalSignatures where
 typeTeamFeatureName :: Doc.DataType
 typeTeamFeatureName = Doc.string . Doc.enum $ cs . toByteString' <$> [(minBound :: TeamFeatureName) ..]
 
+----------------------------------------------------------------------
+-- TeamFeatureTTLValue
+
+-- Using Word to avoid dealing with negative numbers.
+-- Ideally we would also not support zero.
+-- Currently a TTL=0 is ignored on the cassandra side.
+data TeamFeatureTTLValue
+  = TeamFeatureTTLDays Word
+  | TeamFeatureTTLUnlimited
+  deriving stock (Eq, Show, Generic)
+
+deriving instance ToJSON TeamFeatureTTLValue
+
+deriving instance FromJSON TeamFeatureTTLValue
+
+instance ToHttpApiData TeamFeatureTTLValue where
+  toQueryParam a = case a of
+    TeamFeatureTTLDays d -> T.pack . show $ d
+    TeamFeatureTTLUnlimited -> "unlimited"
+
+instance FromHttpApiData TeamFeatureTTLValue where
+  parseQueryParam = \case
+    "unlimited" -> Right TeamFeatureTTLUnlimited
+    d -> case readEither . T.unpack $ d of
+      Right d' -> Right . TeamFeatureTTLDays $ d'
+      Left e -> Left . T.pack $ e <> " oops."
```

Comment:
Errors are currently not surfaced, everything turns into a generic 500. Maybe we should address that. :)

Reply:
but this `e` should result in a 400 generated by servant, no?

in general: yes, if you find an easy way to pass on error objects from galley intact to the stern "UI", please do so.  if it's hard for any reason, please open a ticket.